File: backend/app/src/models/blueprints/Population.py


class Population():
    population_id = 0
    
    def __init__(self, name: str, profession, alive: bool = True):
        Population.population_id += 1
        self.__population_id = Population.population_id
        self.name: str = name
        self.profession: str = profession
        self.alive: bool = alive
    
    def get_population_id(self) -> int:
        return self.__population_id
    
    def fetch_population_state(self) -> dict:
        return {
            "population_id": self.__population_id,
            "category": "population",
            "name": self.name,
            "alive": self.alive
        }
    
    
    

File: backend_refactoring/CSharpApi/Models/BluePrints/Beings/Population.cs
using Microsoft.EntityFrameworkCore;

namespace CSharpApi.Models.BluePrints.Beings
{
    public class Population
    {
        public static int PopulationId { get; set; } = 0;

        public int Id { get; set; }
        
        public string Name { get; set; } = string.Empty;

        public string Profession { get; set; } = string.Empty;

        public bool Alive { get; set; }

        public bool Working { get; set; }

        public string FieldOfWork { get; set; } = string.Empty;

        public bool Employed { get; set; }


        public Population(string name, string profession , bool alive = true)
        {
            PopulationId++;
            Id = PopulationId;
            Name = name;
            Profession = profession;
            Alive = alive;
        }

        public Dictionary<string, object> FetchPopulationState()
        {
            return new Dictionary<string, object> {
                { "population_id", Id},
                { "category", "population" },
                { "name", Name },
                { "profession", Profession },
                { "alive", Alive } };
        }

        public int GetPopulationId() { return Id; }

    }    
}